### PR TITLE
fix: point followed number in css url function be collapsed

### PR DESCRIPTION
**Pull request by jasonxia23**
https://github.com/prettier/prettier/issues/5654

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-css/parser-postcss.js [hunk 1]
```diff
@@ -40,8 +40,17 @@ function parseValueNodes(nodes) {
...
       node.group.groups[0].groups[1].value === ":";
+    const isUnquotedURLWithNumber =
+      node.type === "func" &&
+      node.value === "url" &&
+      node.group &&
+      node.group.groups &&
+      node.group.groups[0] &&
+      node.group.groups[0].groups &&
+      node.group.groups[0].groups.length > 2 &&
+      node.group.groups[0].groups.find(group => group.type === "number");
```
Comment (evilebottnawi) [inline src/language-css/parser-postcss.js]: I think better implement utils `isValidURL` and print all valid url in one line, it is solve many problem, not only with numbers, anyway good work!

Comment (jasonxia23) [reply src/language-css/parser-postcss.js]: Good point. Could you please figure out any other cases?

Comment (jasonxia23) [reply src/language-css/parser-postcss.js]: @evilebottnawi You mean extract `isUnquotedDataURLCall` and `isUnquotedURLWithNumber` to `isValidURL ` util? Maybe I haven't get your point.

Comment (evilebottnawi) [reply src/language-css/parser-postcss.js]: @jasonxia23 i thinks we need get logic from here https://nodejs.org/docs/latest/api/url.html

> A TypeError will be thrown if the input or base are not valid URLs. Note that an effort will be made to coerce the given values into strings. For instance:

```js
const myURL = new URL({ toString: () => 'https://example.org/' });
// https://example.org/
```

And it is valid url we stringify group

Comment (jasonxia23) [reply src/language-css/parser-postcss.js]: Well, I think CSS `<url>` is not same as Javascript `URL`.

For example, I can use `url(foo.png)` but `foo.png` is not "VALID" for `URL`

Comment (lydell) [reply src/language-css/parser-postcss.js]: Technically, at the parsing level the stuff inside `url()` does not have to be a valid URL: https://www.w3.org/TR/css-syntax-3/#url-unquoted-diagram

Comment (evilebottnawi) [reply src/language-css/parser-postcss.js]: @jasonxia23 let's use logic for check what @lydell write above, it is good solution

Comment (jasonxia23) [reply src/language-css/parser-postcss.js]: @evilebottnawi Maybe I should turn to `postcss-values-parser`. Prettier uses `postcss-values-parser` other than `postcss-value-parser`.

**Comment by fisker** [comment]
Thank you for your contribution, the original issue is fixed by #7592.

Your tests added in #8482.

**Pull request closed without merging**